test/exchanges/CCXTExchangeTest.ts

```diff
@@ -0,0 +1,51 @@
+/**********************************************************************************************************************
+ * @license                                                                                                           *
+ * Copyright 2017 Coinbase, Inc.                                                                                      *
+ *                                                                                                                    *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     *
+ * with the License. You may obtain a copy of the License at                                                          *
+ *                                                                                                                    *
+ * http://www.apache.org/licenses/LICENSE-2.0                                                                         *
+ *                                                                                                                    *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on*
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the                 *
+ * License for the specific language governing permissions and limitations under the License.                         *
+ **********************************************************************************************************************/
+
+const assert = require('assert');
+
+import { ExchangeAuthConfig } from '../../src/exchanges/AuthConfig';
+import CCXTExchangeWrapper from '../../src/exchanges/ccxt';
+import { NullLogger } from '../../src/utils/Logger';
+
+describe('CCXT Exchange Wrapper', () => {
+    const exchangeId = 'bitmex';
+    const productId = 'BTC-USD';
+    let wrapper: CCXTExchangeWrapper;
+
+    before(async () => {
+        const auth: ExchangeAuthConfig = { key: null, secret: null };
+        wrapper = CCXTExchangeWrapper.createExchange(exchangeId, auth, NullLogger);
+    });
+
+    it('is able to fetch historical trade data from an exchange', async () => {
+        const data = await wrapper.fetchHistTrades(productId);
```

Comment:
I love the tests, but it looks like these requests are hitting the internet. This leads to slow and flaky tests (what if bitmex is down / delist the product?).  We use the `nock` library for mocking out HTTP requests, and you can see examples of its use in some of the other tests here.

Reply:
> delist the product

That's why don't hardcode pairs and fetch them online where possible...  Outdated product lists can be painful to maintain in the long term. Though we still have to do `unique product id ←→ common symbol` mapping to remain consistent across exchanges... Did you know that CoinMarketCap lists BTM for two different coins (Bytom (HitBtc) vs Bitmark (Poloniex), https://github.com/ccxt-dev/ccxt/issues/348, https://github.com/ccxt-dev/ccxt/issues/350, etc...) ? There's plenty of that across exchanges...

Just my 2 cents.

Reply:
Interesting. Then there's BCH and BCC..

For the GTT, which is an abstraction layer, I think it's ok to maintain a canonical list of coin names and map them behind the interfaces.
So for example, if GTT users define and expect BCC to represent 'Bitcoin cash' then mappings to BCH should happen behind the scenes where appropriate. Then users only have to learn one set of ticker names and the GTT does most of the work. ccxt, being pretty close to the apis needs to work with this more carefully.

Reply:
@CjS77 we do the BCC → BCH mapping, and we use BCH everywhere in our API because BCH is most widely used for Bitcoin Cash, while BCC also stands for Bitconnect sometimes, that causes confusion. If you notice BCC somewhere in CCXT let us know, but you should expect unified symbols from CCXT as written here: https://github.com/ccxt-dev/ccxt/wiki/Manual#naming-consistency ;)

Reply:
Yeah, so we're advocating the same thing. I was just using BCH/BCC as an example and wasn't favoring one ticker over the other.